### ISSUE TITLE
[cdh5] bcprov-jdk14 excluded recently in main pom.xml file causes cermine failure

### ISSUE THREAD

**Issue by marekhorst** [openaire/iis#110]
Apparently in recent **Merge branch 'master' into cdh5** commit https://github.com/openaire/iis/commit/1cc67817a494423b452ac6179aff792201bcc584 the following dependency was excluded from `cermine-impl`:
```
<exclusion>
    <groupId>org.bouncycastle</groupId>
    <artifactId>bcprov-jdk14</artifactId>
</exclusion>
```
which causes cermine module failure:
```
Error: java.lang.ClassNotFoundException: org.bouncycastle.asn1.ASN1OctetString
```

What was the reason of excluding this dependency? I tracked all the changes in main `pom.xml` file history but couldn't find original commit introducing this change.

**Comment by madryk** [comment]
That library was causing problems to connect with frontend machine through ssh.
I was getting some security exceptions. Apparently there can't be two libraries at the same time:
`bcprov-jdk14` and `bcprov-jdk15on`. The last one is included be ssh library which we use (sshj)

Could you point to workflow which failed because of this?

**Comment by marekhorst** [comment]
Here are the failed jobs from the run executed on rumcajs cluster:

http://spark-m1.vls.icm.edu.pl:19888/jobhistory/attempts/job_1438689608784_33291/m/FAILED

The workflow which failed is the metadataextraction one:

https://github.com/openaire/iis/blob/cdh5/iis-workflows/iis-workflows-metadataextraction/src/main/resources/eu/dnetlib/iis/workflows/metadataextraction/oozie_app/workflow.xml

The problem is `bcprov-jdk14` was defined at `compile` scope in `cermine-impl` dependency while `bcprov-jdk15on` is defined as `test` so it is not included in oozie package.

If we are stuck on `bcprov` version clash we can discuss this issue with Dominika 

@dtkaczyk do you think it is possible to upgrade `bcprov` dependency in:

https://github.com/CeON/CERMINE/blob/master/cermine-impl/pom.xml

`bcprov-jdk14` seems to be pretty old one.

**Comment by dtkaczyk** [comment]
I can change bcprov-jdk14 to bcprov-jdk15on version 1.49. Will this be enough to solve the problem?

**Comment by marekhorst** [comment]
> I can change bcprov-jdk14 to bcprov-jdk15on version 1.49. Will this be enough to solve the problem?

I guess this should be enough. @madryk What do you think?

**Comment by madryk** [comment]
It should be fine.

**Comment by marekhorst** [comment]
> It should be fine.

@dtkaczyk OK, you can proceed then.

**Comment by dtkaczyk** [comment]
It's done. New snapshot should be visible in the repository soon.

**Comment by marekhorst** [comment]
@madryk Can you revert dependency exclusion from `pom.xml` file?

**Comment by madryk** [comment]
Issue was fixed by #138
